esx_service/vmdk_ops.py

```diff
@@ -578,11 +573,11 @@ def executeRequest(vm_uuid, vm_name, config_path, cmd, full_vol_name, opts):
         return err(str(ex))
     if not datastore:
         datastore = vm_datastore
-    elif datastore not in known_datastores():
+    elif not vmdk_utils.validate_datastore(datastore):
         return err("Invalid datastore '%s'.\n" \
-                   "Known datastores: %s.\n" \
-                   "Default datastore: %s" \
-                   % (datastore, ", ".join(known_datastores()), vm_datastore))
+                "Known datastores: %s.\n" \
+                "Default datastore: %s" \
+                % (datastore, ", ".join([i[0] for i in vmdk_utils.get_datastores()]), vm_datastore))
```

Comment:
`[i[0] for i in vmdk_utils.get_datastores()` is cut-n-pasted twice, I think it's a small functoins (get_datastore_names_list())